test/SPI/spi-only-import-exportability.swift

```diff
@@ -0,0 +1,152 @@
+/// Test @_spiOnly exportability type-checking
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+/// Generate dependencies.
+// RUN: %target-swift-frontend -emit-module %t/PublicLib.swift \
+// RUN:   -module-name PublicLib -emit-module-path %t/PublicLib.swiftmodule \
+// RUN:   -swift-version 5 -enable-library-evolution
+// RUN: %target-swift-frontend -emit-module %t/SPIOnlyImportedLib.swift \
+// RUN:   -module-name SPIOnlyImportedLib \
+// RUN:   -emit-module-path %t/SPIOnlyImportedLib.swiftmodule \
+// RUN:   -swift-version 5 -enable-library-evolution -I %t
+
+/// Test the client.
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t -verify \
+// RUN:   -experimental-spi-only-imports
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t -verify \
+// RUN:   -experimental-spi-only-imports \
+// RUN:   -enable-library-evolution
+
+/// Generate the swiftinterface of the working code and verify it.
+// RUN: %target-swift-emit-module-interface(%t/Client.swiftinterface) \
+// RUN:   %t/Client.swift -I %t -DSKIP_ERRORS \
+// RUN:   -experimental-spi-only-imports
+// RUN: %target-swift-typecheck-module-from-interface(%t/Client.swiftinterface) -I %t
```

Comment:
Should we also generate and type check the private interface?

Reply:
I had not thought about private interfaces when creating the `%target-swift-emit-module-interface` lit substitution; we may want another for convenience.

Reply:
Good point. Added it as `%target-swift-emit-module-interfaces(Main.swiftinterface, Main.private.swiftinterface)`. I kept the verify substitution as separate because it's likely to fail so I think it will be easier to debug that way.

Reply:
We could also keep only the current one `%target-swift-emit-module-interface` with a single argument and change the regex to then insert a `.private.` in the path. Do you have a preference for one or another?

Reply:
I think I like the two argument version best because it's a little easier to piece together what's happening in the `RUN:` lines with that context, even if it is a bit more verbose.